util/file.go

```diff
@@ -76,12 +76,51 @@ func CanonicalPath(path string, basePath string) (string, error) {
 	}
 	absPath, err := filepath.Abs(path)
 	if err != nil {
-		return "", err
+		return "", errors.WithStackTrace(err)
 	}
 
 	return CleanPath(absPath), nil
 }
 
+// GlobCanonicalPath returns the canonical versions of the given glob paths, relative to the given base path.
+func GlobCanonicalPath(basePath string, globPaths ...string) ([]string, error) {
+	if len(globPaths) == 0 {
+		return []string{}, nil
+	}
+
+	var err error
+
+	// Ensure basePath is cannonical
+	basePath, err = CanonicalPath("", basePath)
+	if err != nil {
+		return nil, err
+	}
+
+	var paths []string
+
+	for _, globPath := range globPaths {
+		// Ensure globPath are absolute
+		if !filepath.IsAbs(globPath) {
+			globPath = filepath.Join(basePath, globPath)
+		}
+
+		matches, err := zglob.Glob(globPath)
```

Comment:
What is the desired behavior here if `err` is not nil?

Reply:
Just skip this path as it was before
https://github.com/gruntwork-io/terragrunt/blob/40e0d4b5db0fcfae9d9d36e3430995b7d142d800/configstack/module.go#L182-L187

Perhaps there is a reason for this behaviour, I don't know, but I would check the correctness of this path and return an error in cases of non-existence.

Reply:
Do you think we should change this behaviour?

Reply:
Makes sense, no worries! Let's leave it for now